eth/api_backend.go

```diff
@@ -74,7 +74,7 @@ func (b *EthApiBackend) StateAndHeaderByNumber(blockNr rpc.BlockNumber) (ethapi.
 	// Pending state is only known by the miner
 	if blockNr == rpc.PendingBlockNumber {
 		block, state := b.eth.miner.Pending()
-		return EthApiState{state}, block.Header(), nil
+		return EthApiState{state.Copy()}, block.Header(), nil
```

Comment:
It seems it _is_ making a copy there. I can't explain why that doesn't fix the issue, only confirm that adding the copy here definitely does.